src/main/java/com/imaee/propinq/users/data/repositories/ITokenRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.imaee.propinq.users.data.repositories;
+
+import com.imaee.propinq.users.data.models.Token;
+import com.imaee.propinq.users.data.models.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public interface ITokenRepository extends JpaRepository<Token, UUID> {
+    @Query("SELECT t FROM tokens t " +
+            "WHERE t.user = ?1 " +
+            "AND t.tokenExpirationDate > CURRENT_TIMESTAMP ")
+    Optional<Token> findActiveTokenByUser(User user);
```

Comment:
Bien. Podríamos guardar las queries en la parte `/utils` del package, quizás. Para más limpieza en el código

```
public class TokenQueries {
        public static final String ACTIVE_TOKEN_BY_USER_AND_EXPIRATION = "SELECT t FROM tokens t " + 
                "WHERE t.user = ?1 AND t.tokenExpirationDate > CURRENT_TIMESTAMP";
}
```

```suggestion
    @Query(ACTIVE_TOKEN_BY_USER_AND_EXPIRATION)
    Optional<Token> findActiveTokenByUser(User user);
```